google/auth/transport/requests.py

```diff
@@ -154,7 +155,9 @@ def __init__(self, credentials,
         # credentials.refresh).
         # Do not pass `self` as the session here, as it can lead to infinite
         # recursion.
-        self._auth_request = Request()
+        sess = requests.Session()
+        sess.mount("https://", requests.adapters.HTTPAdapter(max_retries=3))
```

Comment:
Could you add a comment here as to why we do this? I think that'll be really useful for future maintainers.